src/User/Application/CommandHandler/RequestPasswordResetCommandHandler.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\User\Application\CommandHandler;
+
+use App\Shared\Domain\Bus\Command\CommandHandlerInterface;
+use App\Shared\Domain\Bus\Event\EventBusInterface;
+use App\User\Application\Command\RequestPasswordResetCommand;
+use App\User\Application\Command\RequestPasswordResetCommandResponse;
+use App\User\Domain\Entity\PasswordResetTokenInterface;
+use App\User\Domain\Entity\UserInterface;
+use App\User\Domain\Event\PasswordResetRequestedEvent;
+use App\User\Domain\Exception\PasswordResetRateLimitExceededException;
+use App\User\Domain\Factory\PasswordResetTokenFactoryInterface;
+use App\User\Domain\Repository\PasswordResetTokenRepositoryInterface;
+use App\User\Domain\Repository\UserRepositoryInterface;
+use Symfony\Component\Uid\Factory\UuidFactory;
+
+final readonly class RequestPasswordResetCommandHandler implements
+    CommandHandlerInterface
+{
+    public function __construct(
+        private UserRepositoryInterface $userRepository,
+        private PasswordResetTokenRepositoryInterface $tokenRepository,
+        private PasswordResetTokenFactoryInterface $tokenFactory,
+        private EventBusInterface $eventBus,
+        private UuidFactory $uuidFactory,
+        private int $rateLimitMaxRequests = 3,
+        private int $rateLimitWindowHours = 1,
+    ) {
+    }
+
+    public function __invoke(RequestPasswordResetCommand $command): void
+    {
+        $successMessage = 'If valid, you will receive a password reset link.';
+
+        $user = $this->userRepository->findByEmail($command->email);
+
+        if (!$user instanceof UserInterface) {
+            $command->setResponse(
+                new RequestPasswordResetCommandResponse($successMessage)
+            );
+            return;
+        }
+
+        $this->checkRateLimit($command->email);
+        $token = $this->createPasswordResetToken($user);
+        $this->publishEvent($user, $token);
+
+        $command->setResponse(
+            new RequestPasswordResetCommandResponse($successMessage)
+        );
+    }
+
+    private function checkRateLimit(string $email): void
+    {
+        $since = new \DateTimeImmutable(
+            "-{$this->rateLimitWindowHours} hours"
+        );
+        $recentRequests = $this->tokenRepository->countRecentRequestsByEmail(
```

Comment:
rate limiting information should be stored in the cache using symfony cache instead of using the database and the table for tokens since it violates the single SRP from SOLID